ecr-login/ecr.go

```diff
@@ -70,14 +71,37 @@ func NewECRHelper(opts ...Option) *ECRHelper {
 // ensure ECRHelper adheres to the credentials.Helper interface
 var _ credentials.Helper = (*ECRHelper)(nil)
 
-func (ECRHelper) Add(creds *credentials.Credentials) error {
-	// This does not seem to get called
-	return notImplemented
+func shouldIgnoreCredsStorage() bool {
+	return os.Getenv("AWS_ECR_IGNORE_CREDS_STORAGE") == "true"
 }
 
-func (ECRHelper) Delete(serverURL string) error {
-	// This does not seem to get called
-	return notImplemented
+// Add tries to store credentials when docker requests it. This usually happens during `docker login` calls. In our context,
+// storing arbitrary user given credentials makes no sense.
+func (self ECRHelper) Add(creds *credentials.Credentials) error {
+	if shouldIgnoreCredsStorage() {
+		self.logger.
+			WithField("username", creds.Username).
+			WithField("serverURL", creds.ServerURL).
+			Warning("Ignoring request to store credentials. " +
+				"This is not supported in the context of the docker ecr-login helper.")
```

Comment:
Wouldn't it be more useful to output this error message when `shouldIgnoreCredsStorage()` is not true? and add something like

```
To ignore this error, set env value `AWS_ECR_IGNORE_CREDS_STORAGE` to true
```

Reply:
We're interested in getting this change merged. I am picking up this PR through here: https://github.com/awslabs/amazon-ecr-credential-helper/pull/847 so we can get feedback addressed and hopefully merged soon. Can you please relay feedback for this PR there and I can look at resolving these.

In the meantime:
Can you elaborate on what you mean @swagatbora90? I am not quite sure on what the intention of this suggestion is.

Reply:
Upon thinking about it for longer, I think I understand. I will implement this in my draft.

Reply:
My understanding is that we're preserving the `notImplemented` error, and this particular behaviour is what users want to avoid when all they want to do is authenticate, whether or not storing the credential is supported doesn't matter. They want to get the auth done.

The environment variable is meant to provide a means to skip the attempt at storing the credential and throw a non-blocking warning rather than an error. If we inverted the condition for checking `shouldIgnoreCredsStorage`, users will still get the `notImplemented` error and doesn't satisfy what the change was intending to do. The test `TestAddIgnored` reflects this behaviour.